### PR TITLE
Cannot set property 'gte' of undefined #80

### LINKED ISSUE #80: Cannot set property 'gte' of undefined
image

This error occurs for Elasticsearch because of we use hardcoded date field name instead of name from datasource structure:
https://github.com/CorpGlory/grafana-datasource-kit/blob/master/src/metrics/elasticsearch_metric.ts#L25

### PR DESCRIPTION
**Pull request by amper43**
fix #80 

### Changes
- use time field from `datasource` structure instead of using `@timestamp`

### CODE REVIEW

File: src/metrics/elasticsearch_metric.ts [hunk 1]
```diff
@@ -16,19 +16,35 @@ export class ElasticsearchMetric extends AbstractMetric {
     }
 
     data[1].size = 0;
+    let timeField = null;
+
+    let aggs = _.filter(data[1].aggs, f => _.has(f, 'date_histogram'));
+    _.each(aggs, agg => {
+      agg.date_histogram.extended_bounds = {
+      min: from.toString(),
```
Comment (rozetko) [inline src/metrics/elasticsearch_metric.ts]: Indentation

File: src/metrics/elasticsearch_metric.ts [hunk 2]
```diff
@@ -16,19 +16,35 @@ export class ElasticsearchMetric extends AbstractMetric {
...
+      max: to.toString()
+      };
+
+      if(timeField !== null) {
+        console.warn(
+          `got more than one datasource time filed, change ${timeField} to ${agg.date_histogram.field}`
```
Comment (rozetko) [inline src/metrics/elasticsearch_metric.ts]:
```suggestion
          `got more than one datasource time field, change ${timeField} to ${agg.date_histogram.field}`
```

File: src/metrics/elasticsearch_metric.ts [hunk 3]
```diff
@@ -16,19 +16,35 @@ export class ElasticsearchMetric extends AbstractMetric {
     }
 
     data[1].size = 0;
```
Comment (jonyrock) [inline src/metrics/elasticsearch_metric.ts]: what is `data[1]` ? please declare a var for this

File: src/metrics/elasticsearch_metric.ts [hunk 4]
```diff
@@ -16,19 +16,35 @@ export class ElasticsearchMetric extends AbstractMetric {
...
+    let timeField = null;
+
+    let aggs = _.filter(data[1].aggs, f => _.has(f, 'date_histogram'));
```
Comment (jonyrock) [inline src/metrics/elasticsearch_metric.ts]: lets move `'date_histogram'` to  constant

File: src/metrics/elasticsearch_metric.ts [hunk 5]
```diff
@@ -16,19 +16,35 @@ export class ElasticsearchMetric extends AbstractMetric {
...
+    _.each(aggs, agg => {
```
Comment (jonyrock) [inline src/metrics/elasticsearch_metric.ts]: please add type of `agg`

File: src/metrics/elasticsearch_metric.ts [hunk 6]
```diff
@@ -10,19 +19,20 @@ export class ElasticsearchMetric extends AbstractMetric {
 
   getQuery(from: number, to: number, limit: number, offset: number): MetricQuery {
     let data = this.datasource.data.split('\n').map(d => d === '' ? d: JSON.parse(d));
-
     if(data.length === 0) {
       throw new Error('Datasource data is empty');
```
Comment (jonyrock) [inline src/metrics/elasticsearch_metric.ts]: maybe throw https://github.com/CorpGlory/grafana-datasource-kit/blob/master/src/grafana_service.ts#L8 here?

File: src/metrics/elasticsearch_metric.ts [hunk 7]
```diff
@@ -10,19 +19,20 @@ export class ElasticsearchMetric extends AbstractMetric {
...
     }
 
-    data[1].size = 0;
+    const queryConfig = data[1];
```
Comment (jonyrock) [inline src/metrics/elasticsearch_metric.ts]: let's be specific about what type do you expect here

File: src/metrics/elasticsearch_metric.ts [hunk 8]
```diff
@@ -10,19 +19,20 @@ export class ElasticsearchMetric extends AbstractMetric {
...
+
+    queryConfig.size = 0;
     let timeField = null;
 
-    let aggs = _.filter(data[1].aggs, f => _.has(f, 'date_histogram'));
-    _.each(aggs, agg => {
+    let aggs = _.filter(queryConfig.aggs, f => _.has(f, 'date_histogram'));
```
Comment (jonyrock) [inline src/metrics/elasticsearch_metric.ts]: let's make a constant for this literal somewhere on the top of the file

File: src/metrics/elasticsearch_metric.ts [hunk 9]
```diff
@@ -10,19 +19,20 @@ export class ElasticsearchMetric extends AbstractMetric {
...
+    _.each(aggs, (agg: Aggregation) => {
       agg.date_histogram.extended_bounds = {
```
Comment (jonyrock) [inline src/metrics/elasticsearch_metric.ts]: let's use `'date_histogram'` literal as `agg[LITERAL_CONSTANT]`

**Pull request merged**